pkg/api/networkservice/connectioncontext.proto

```diff
@@ -79,9 +79,17 @@ message EthernetContext {
     string dst_mac = 2;
 }
 
+message SRIOVContext {
```

Comment:
Why is this in connection context?  Its a *mechanism* ... it should never leave the local server...

Reply:
There is actually a connection established between client and endpoint - client VF to endpoint VF. As we have planned to implement it, forwarder shares it's SR-IOV state with the endpoint, endpoint select it's own VF and PF + IOMMU group for the forwarder and so forwarder selects VF for the client on its own side.
So these `EndpointVfPciAddress, ClientPfPciAddress, ClientVfPciAddress, IommuGroup` we need to establish connection between client and endpoint, it is not something local for the host as device numbers and cgroup directory path for VFIO or net interface name as for kernel.

Reply:
All of that is fine... but its *still* a mechanism level thing returned for a vfio mechanism from the forwarder as a mechanism... *not* a ConnectionContext level thing.

Reply:
Is is not only for **vfio** mechanism - it is for all mechanisms supported in SR-IOV, because it is **not for mechanism**, it is for **VF - VF** pair which is actually the connection.

Reply:
So for any SR-IOV mechanism: kernel, VFIO or probably something else we can invent, there is a common thing we need to establish the connection - on both sides we need to take a PCI function and mark it as bounded to the **connection**.
Here is a common scheme we have planned for the connection to be established:
```
kernel case  Client     ---{ mechanism }--->                                        Forwarder
VFIO case    Client     ---{ mechanism, cgroup directory }--->                      Forwarder
             Forwarder  ---{ ..., SR-IOV state }--->                                Endpoint
             Endpoint   ---{ ..., endpoint VF, client PF, client IOMMU group }--->  Forwarder
kernel case  Forwarder  ---{ ..., client VF, net interface }--->                    inject.NewServer()
VFIO case    Forwarder  ---{ ..., client VF, device numbers }--->                   vfio.NewServer()
```
So the things need for all types of SR-IOV connections are:
* mechanism - stored in Connection as Mechanism
* SR-IOV state - stored in Connection.Context.ExtraContext as YAML string
* endpoint VF, client PF, client IOMMU group, client VF - stored in Connection.Context.SRIOVContext

The things we need for the kernel connection:
* net interface

The things we need for the VFIO connection:
* cgroup directory
* device numbers

As you can see, all common things are mostly used on Forwarder-Endpoint interaction level, so it is not something local.
Our approach is to put common things into the Connection and put kernel/VFIO specific stuff into the related Mechanism.

Reply:
In the VFIO case, the Forwarder receive the Mechanism sent by the Client (its simply passed through the NSMgr) and so any SR-IOV state can be kept in the mechanism and still accessed by the Forwarder.  There is zero reason for the Endpoint to receive any SRIOV information, not to return any SRIOV Information to the Forwarder.  The SRIOV information needs to only be shared, via the mechanism, between the Client and the Forwarder.